pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1380,9 +1380,18 @@ func (l *LibvirtDomainManager) ListAllDomains() ([]*api.Domain, error) {
 	return list, nil
 }
 
-func (l *LibvirtDomainManager) setDomainSpecWithHooks(vmi *v1.VirtualMachineInstance, spec *api.DomainSpec) (cli.VirDomain, error) {
+func (l *LibvirtDomainManager) setDomainSpecWithHooks(vmi *v1.VirtualMachineInstance, origSpec *api.DomainSpec) (cli.VirDomain, error) {
 
+	spec := origSpec.DeepCopy()
 	hooksManager := hooks.GetManager()
+
+	if val := vmi.Annotations[v1.PlacePCIDevicesOnRootComplex]; val == "true" {
+		err := api.PlacePCIDevicesOnRootComplex(spec)
```

Comment:
why isn't this done in the `Convert_v1_VirtualMachine_To_api_Domain()` function instead?

This technically works the way you have it, however I'd be surprised to find out the domain is getting mutated during the "set" function like this.

Reply:
Addressed.